job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java

```diff
@@ -78,13 +96,19 @@ private boolean shouldIgnore(URIStatus status) {
     AlluxioURI outputDir = new AlluxioURI(config.getOutput());
 
     List<URIStatus> files = Lists.newArrayList();
+    long sum = 0;
     for (URIStatus status : context.getFileSystem().listStatus(inputDir)) {
       if (!shouldIgnore(status)) {
         files.add(status);
+        sum += status.getLength();
       }
     }
     Map<WorkerInfo, ArrayList<CompactTask>> assignments = Maps.newHashMap();
-    int groupSize = Math.max(1, (files.size() + 1) / config.getNumFiles());
+    int numOfFiles = config.getNumFiles();
```

Comment:
I feel like the file size should also be a config parameter.

Reply:
I am not sure which file size you are referring to here? the file size can not be determined until we read the result of listStatus right? that's why it can not be a config parameter?

Reply:
What I meant was, when we do something like `averageSize < FileUtils.ONE_GB`, we should use the file size as a config parameter.